examples/driver.rs

```diff
@@ -11,7 +12,9 @@ fn main() {
     let mut op = compio::op::ReadAt::new(file.as_raw_fd(), 0, Vec::with_capacity(4096));
     unsafe { driver.push(&mut op, 0) }.unwrap();
 
-    let entry = driver.poll_one(None).unwrap();
+    let mut entries = ArrayVec::<Entry, 1>::new();
+    driver.poll(None, &mut entries).unwrap();
```

Comment:
Using arrays is nice in terms of autovectorization opportunities. ArrayVec adds minor runtime overhead.

Maybe we could let users to do the choice in compile time and provide two methods - poll and poll_boxed?

Reply:
What do you mean by overhead?

Reply:
It's weird to provide a method `poll_boxed`. The users should *not* be encouraged to heap alloc. A recommended solution is to use `SmallVec`. However in this example, an `ArrayVec` is enough.

Reply:
The idea is to allocate on heap only during application startup.

Maybe we could define a public Array trait and implement it for stack-based array and Box<[Entry; N]>?

The interface will be generic and even support use cases when user embeds an array into larger heap allocated structure.

Reply:
I think the current interface with `impl Extend` is good enough and very universal.